TombEngine/Scripting/Internal/TEN/Objects/Moveable/MoveableObject.cpp

```diff
@@ -1016,3 +1047,104 @@ bool Moveable::MeshExists(int index) const
 
 	return true;
 }
+
+//Attach camera and camera target to a mesh of an object.
+void  Moveable::AttachCamera(short camMeshId, short targetMeshId)
+{
+	ObjCamera(m_item, camMeshId, m_item, targetMeshId, true);
+}
+
+//Lara will undraw her weapons if they are drawn, throw away flare if she helds one in her hand.
+void   Moveable::UndrawWeapons()
+{
+	if (m_item->IsLara())
+	{
+		auto* lara = GetLaraInfo(m_item);
+
+		if (lara->Control.HandStatus != HandStatus::Free ||
+			lara->Control.Weapon.GunType == LaraWeaponType::Flare)
+		{
+			lara->Control.HandStatus = HandStatus::WeaponUndraw;
+		}
+	}
+}
+
+//Lara will throw away the torch if she helds one in her hand.
+void   Moveable::ThrowawayTorch()
+{
+	if (m_item->IsLara())
+	{
+		auto* lara = GetLaraInfo(m_item);
+
+		if (lara->Control.Weapon.GunType == LaraWeaponType::Torch)
+		{
+			Lara.Torch.State = TorchState::Dropping;
+		}
+	}
+}
+
+//Returns 0=HandsFree, 1=Busy(climbing,etc), 2=WeaponDraw, 3=WeaponUndraw, 4=WeaponInHand.
+HandStatus Moveable::GetLaraHandStatus() const
+{
+	if (m_item->IsLara())
```

Comment:
As above

Reply:
Changed to make it always Lara Item. No check needed anymore :)